x-pack/plugins/security_solution/public/common/components/discover_in_timeline/use_discover_in_timeline_actions.tsx

```diff
@@ -205,6 +205,10 @@ export const useDiscoverInTimelineActions = (
       } else {
         // If no saved search exists. Create a new saved search instance and associate it with the timeline.
         try {
+          // Make sure we're not creating a saved search while a previous creation call is in progress
+          if (status !== 'idle') {
```

Comment:
isnt there some kind of enum we could use here?

Reply:
@lgestc `status` comes from `@tanstack/query-core` and is union of string literals:

```ts
export type MutationStatus = 'idle' | 'loading' | 'success' | 'error'
```

So the type is checked, just not as an enum :)